src/main/kotlin/org/rust/ide/refactoring/extractEnumVariant/RsExtractEnumVariantProcessor.kt

```diff
@@ -54,9 +54,16 @@ class RsExtractEnumVariantProcessor(
         val parameters = filterTypeParameters(element.typeReferences, enum.typeParameterList)
         val typeParametersText = parameters.format()
         val whereClause = buildWhereClause(enum.whereClause, parameters)
+        val attributes = findTransitiveAttributes(ctx, TRANSITIVE_ATTRIBUTES)
 
         val struct = element.createStruct(enum.vis?.text, name, typeParametersText, whereClause)
         val inserted = enum.parent.addBefore(struct, enum) as RsStructItem
+
+        for (attribute in attributes) {
+            val copiedAttribute = factory.createOuterAttr(attribute.metaItem.text)
+            inserted.parent.addBefore(copiedAttribute, inserted)
```

Comment:
The reason of extra whitespace is here - you insert new attributes *before* struct itself although outer attributes should be part of struct item. As a result, you have attributes without parent struct and struct without attributes that leads to whitespace between them in editor after formatting

The existing code can be fixed in the following way `inserted.addBefore(copiedAttribute, inserted.firstChild)` but I'd recommend avoiding this approach here because each call of `factor.create*` creates a temporary file, fully parses it, and extracts necessary PSI nodes.

We already create struct item via psi factory. Let's add text of necessary attributes into initial struct text and create fully "configured" struct node by `element.createStruct` call, i.e. single `factor.create*` invocations

Reply:
Thanks, now it makes sense.